### PR TITLE
Ansible playbook details and schedule page

### PR DESCRIPTION
**Pull request by cbbayburt**
Implement Ansible playbook details page to show a playbook's content and schedule execution.

## GUI diff
playbook-details

## Test coverage
- Unit tests will be added with a separate PR

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: web/html/src/manager/minion/ansible/schedule-playbook.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  const [inventories, setInventories] = useState<string[]>([]);
+  const [actionChain, setActionChain] = useState<ActionChain | null>(null);
+  const [datetime, setDatetime] = useState(new Date());
+
+  // TODO: Use UserLocalization hook instead
+  const timezone = window.timezone;
+  const localTime = window.localTime;
+
+  useEffect(() => {
+    const getInventoryPaths = async () => {
```
Comment (Etheryte) [inline web/html/src/manager/minion/ansible/schedule-playbook.tsx:34]: Small nitpick, here and elsewhere, if the whole async flow is handled by chaining `then`, `catch` etc then making the function `async` doesn't offer any additional benefit.

**Review by Etheryte — APPROVED** [review]
Looks good. 👍

File: web/html/src/manager/minion/ansible/schedule-playbook.tsx (hunk 1 above)
Comment (cbbayburt) [reply web/html/src/manager/minion/ansible/schedule-playbook.tsx:34]: oh right, the chain already returns a `Promise`. I'll remove `async` :+1:

File: web/html/src/manager/minion/ansible/schedule-playbook.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      .then((res: JsonResult<number>) => res.success ? res.data : Promise.reject(res))
+      .then(actionId =>
+        setMessages(MsgUtils.info(<ScheduleMessage id={actionId} actionChain={actionChain?.text}/>)))
+      .catch(res =>
+        setMessages(res.messages?.flatMap(MsgUtils.error) || Network.responseErrorMessage(res))
+      );
+  };
+
+  if (loading)
+    return (<p>{t("Loading playbook contents...")}</p>); //TODO: Make pretty
```
[the remark this answers is not in the document]
Comment (ncounter) [reply web/html/src/manager/minion/ansible/schedule-playbook.tsx]: You just need to
`import { Loading } from "components/utils/Loading";`
...
`<Loading text={t("Loading playbook contents...")} />`

[the remark this answers is not in the document]
Comment (hustodemon) [reply web/html/src/manager/minion/ansible/schedule-playbook.tsx]: Already done :smile:

File: web/html/src/manager/minion/ansible/ansible-path-content.tsx [hunk 3]
```diff
@@ -43,6 +46,14 @@ class AnsiblePathContent extends React.Component<PropsType, StateType> {
   }
 
   render () {
+    if (this.state.selectedPlaybook) {
+      return (
+        <SchedulePlaybook
+          playbook={this.state.selectedPlaybook}
+          onBack={() => this.setState({selectedPlaybook: null})} />
```
Comment (ncounter) [inline web/html/src/manager/minion/ansible/ansible-path-content.tsx:53]: Very nice and non invasive injection to swap the page, :+1:

**Review by ncounter — APPROVED** [review]
LGTM :+1:

**Review by hustodemon — APPROVED** [review]
LGTM, thanks!

**Pull request merged**